src/Core/Parser/TemplateProcessor/NamespaceDetectionTemplateProcessor.php

```diff
@@ -125,70 +124,35 @@ public function registerNamespacesFromTemplateSource($templateSource)
                         substr($templateSource, strrpos($templateSource, $closingTag) + strlen($closingTag));
                 }
             } else {
-                if (!empty($namespaces)) {
-                    $namespaceAttributesToRemove = [];
-                    foreach ($namespaces as $namespace) {
-                        if (!$viewHelperResolver->isNamespaceIgnored($namespace[1])) {
-                            $namespaceAttributesToRemove[] = preg_quote($namespace[1], '/') . '="' . preg_quote($namespace[2], '/') . '"';
-                        }
-                    }
-                    if (count($namespaceAttributesToRemove)) {
-                        $matchWithRemovedNamespaceAttributes = preg_replace('/(?:\\s*+xmlns:(?:' . implode('|', $namespaceAttributesToRemove) . ')\\s*+)++/', ' ', $matches[0]);
-                        $templateSource = str_replace($matches[0], $matchWithRemovedNamespaceAttributes, $templateSource);
+                $namespaceAttributesToRemove = [];
+                foreach ($namespaces as $namespace) {
+                    if (!$viewHelperResolver->isNamespaceIgnored($namespace[1])) {
+                        $namespaceAttributesToRemove[] = preg_quote($namespace[1], '/') . '="' . preg_quote($namespace[2], '/') . '"';
                     }
                 }
+                if (count($namespaceAttributesToRemove)) {
+                    $matchWithRemovedNamespaceAttributes = preg_replace('/(?:\\s*+xmlns:(?:' . implode('|', $namespaceAttributesToRemove) . ')\\s*+)++/', ' ', $matches[0]);
```

Comment:
why do we need \s*+ (non-greedy) operators here?

Reply:
This expression has not changed since Fluid was created so it's not exactly related to this change. But, as far as I can determine, this part ensures that `" ` and `">` are both matched (two variations on how an attribute value may end: space or tag end after attr value).

Reply:
Ah right, missed that it was there already. Then leave as is